### PR TITLE
multi user support

### PR DESCRIPTION
**Pull request by AnkRaiza**
Aditional endpoints to support multi user

### CODE REVIEW

File: src/main/firebase/FirebaseController.ts [hunk 1]
```diff
@@ -201,6 +209,103 @@ export class FirebaseDBUser implements FirebaseUserObj {
             return Promise.reject(new Error("User is not a member of the source."));
         }
     }
+
+    addTeamMember(user: any): Promise<FirebaseDBUser> {
+        const teamCopy = Object.assign({}, this.team);
+        teamCopy[user.id] = {userType: user.userType, enableNotifications: false};
+        return this.myRef
```
Comment (jperata) [inline src/main/firebase/FirebaseController.ts]: async await are a lot cleaner and easier to set path if source is not found or something.

const sourcesResult = await this.myRef.child("sources").once("value");
await this.myRef.child("team").set(teamCopy);
return new FirebaseDBUser(this.userId, this.db, { sources: sourcesResult.val(), team: teamCopy });

File: src/main/firebase/FirebaseController.ts [hunk 2]
```diff
@@ -201,6 +209,103 @@ export class FirebaseDBUser implements FirebaseUserObj {
...
+                    .then(() => {
+                        return new FirebaseDBUser(this.userId, this.db, { sources: sourcesResult.val(), team: teamCopy });
+                    });
+            });
+    }
+
+    addMemberToUser(user: any): Promise<FirebaseDBUser> {
+        const memberCopy = Object.assign({}, this.member);
+        memberCopy[user.id] = user.userType;
+        return this.myRef
```
Comment (jperata) [inline src/main/firebase/FirebaseController.ts]: same

File: src/main/firebase/FirebaseController.ts [hunk 3]
```diff
@@ -201,6 +209,103 @@ export class FirebaseDBUser implements FirebaseUserObj {
...
+            .child("member")
+            .set(memberCopy)
+            .then(() => {
+                return new FirebaseDBUser(this.userId, this.db, {});
+            });
+    }
+
+    addTeamSources(user: any): any {
+        return this.myRef
```
Comment (jperata) [inline src/main/firebase/FirebaseController.ts]: same

File: src/main/index.ts [hunk 4]
```diff
@@ -47,6 +47,10 @@ app.get("/v1/sources", Services.getSources(auth, db));
 
 app.get("/v1/sourceId", Services.getSourceId(db));
 
+app.post("/v1/addTeamMember", Services.addTeamMember(auth, db));
+
+app.post("/v1/getTeamById", Services.getTeam(auth, db));
```
Comment (jkelvie) [inline src/main/index.ts]: Why is this not a get? And I would just call it /v1/team

File: src/main/services/v1/GetTeam.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import { auth } from "../auth";
+
+export function getTeam(adminAuth: Admin.auth.Auth, db: Admin.database.Database): (req: Express.Request, res: Express.Response) => Promise<Express.Response> {
+    return auth(async function(req: Express.Request, res: Express.Response): Promise<Express.Response> {
+        const fbDb = new FirebaseDatabase(db);
+        if (!req.query.id) {
+            Returns.NotOkay(res, 400, "user id not found").send();
+            return Promise.resolve(res);
+        }
+        const currentUser: FirebaseDBUser = await fbDb.getUser({userId: req.query.id});
```
Comment (chris-ramon) [inline src/main/services/v1/GetTeam.ts]: Let's wrap here from line 15 to 22, within a try/catch and handle errors as `addTeamMember` :+1:

File: src/test/firebase/MockFirebase.ts [hunk 6]
```diff
@@ -135,6 +135,10 @@ export class AuthMock {
         return (user) ? Promise.resolve(user) : Promise.reject(new Error("User not contained in auth."));
     }
 
+    getUserByEmail(email: string): Promise<{}> {
```
Comment (chris-ramon) [inline src/test/firebase/MockFirebase.ts]: should we keep this function ?

Comment (AnkRaiza) [reply src/test/firebase/MockFirebase.ts]: it's an empty function for now, i use the actual function on my changes and i need this empty to sinon.stub it

**Pull request merged**